go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -707,6 +707,15 @@ func (bnc *BaseNetworkController) assignPodAddresses(switchName string) (net.Har
 		podCIDRs []*net.IPNet
 		err      error
 	)
+
+	if bnc.TopologyType() == ovntypes.Layer2Topology && len(bnc.Subnets()) == 0 {
```

Comment:
why not make use of doesNetworkRequireIPAM() function?

Reply:
Right! I will move that method from `BaseSecondaryNetworkController` to `BaseNetworkController` so I can re-use it across the board. 

Good suggestion. 
... should have seen it myself. 

Thank you.

Reply:
Are you OK with providing these 2 methods to `BaseNetworkController` (thus avoiding using double negation):
```golang
func (bnc *BaseNetworkController) doesNetworkTolerateIPAMLessConfig() bool {
       return bnc.TopologyType() == types.Layer2Topology && len(bnc.Subnets()) == 0
}

func (bnc *BaseNetworkController) doesNetworkRequireIPAM() bool {
       return !bnc.doesNetworkTolerateIPAMLessConfig()
}
```
Please pay special attention to the suggested names.